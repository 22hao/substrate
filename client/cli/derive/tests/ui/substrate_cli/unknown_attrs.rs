use sc_cli_derive::substrate_cli;

struct MyCli {}

#[substrate_cli(
	support_url = "http://example.org",
	copyright_start_year = 2020,
	impl_version = "0.1.0",
	unknown_attrs = 42,
)]
impl sc_cli::SubstrateCli for MyCli {
	fn load_spec(&self, id: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
		unimplemented!()
	}
}

fn main() {}
