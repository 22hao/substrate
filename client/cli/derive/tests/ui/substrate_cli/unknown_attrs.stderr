error: unknown macro parameters: unknown_attrs
  --> $DIR/unknown_attrs.rs:5:1
   |
5  | / #[substrate_cli(
6  | |     support_url = "http://example.org",
7  | |     copyright_start_year = 2020,
8  | |     impl_version = "0.1.0",
9  | |     unknown_attrs = 42,
10 | | )]
   | |__^
